### DISCUSSION TITLE
Warning: There are rows full of zeros in y.

### DISCUSSION THREAD

**Discussion by AlainZuur** [JenniNiku/gllvm#178] [General]
Hello,

Why does glvm throw this warning message if there are rows with a species total of 0?

Warning message:
In gllvm(y = SpecData, X = CovX.s, studyDesign = ExtraVariables,  :
  There are rows full of zeros in y. 

I understand that you need to do this in CA/CCA. But why with a gllvm? A covariate may cause a site to have zero species. And I understand that if you have too many of these, then you may get numerical estimation issues. 

Just a general warning?

Kind regards,
Alain

**Comment by BertvanderVeen** [comment]
Rows - not columns. I.e., the warning is thrown when there are sites full of zero; you cannot estimate the site scores without data for unconstrained LVs, or for random row effects.

**Reply by AlainZuur** [reply]
Okay, I would have thought that the normal distribution on these terms would partly account for that... especially for the random row effects. But I'll take your word for it. 

Is this because it may try to use   "- very big"  for a random row effect in case of a log link function and a row full of zeros? Or is the inverse of the site total used somewhere in the maths?


I can already hear some biologists making noise about this. Environmental conditions may be such that there are no species present at a site. But those can be genuine zeros. 

Alain

**Reply by hrlai** [reply]
Thanks Alain for raising this. I fully understand Bert's explanation, but this remains a question that I couldn't reconcile. For single-species SDM, we would and could keep those sites with zeros (since we do not estimate site/row effects here), but for multiple-species JSDMs, this becomes an inference problem... For site LVs, it also feels like the "double zeros" problem in distance-based ordination. By keeping the sites with all zeros, the sites may be more similar than otherwise, because they share a lot more absences...

**Reply by BertvanderVeen** [reply]
This is **not** the double-zero problem from distance - based ordination, nor is it a similar problem. If included, two zeros of different species at the same site are treated differently due to different mean-variance relationships for those species.

Despite the fact that the site effects are random, they still need to be estimated as a function of the data, which is impossible to do accurately if there is no information for a site.

That is not to say you _cannot_ include them. I just do not think it is sensible, and it is likely to lead to convergence issues.
